fvm/transactionContractFunctionInvocator.go

```diff
@@ -0,0 +1,73 @@
+package fvm
+
+import (
+	"fmt"
+
+	"github.com/opentracing/opentracing-go"
+	traceLog "github.com/opentracing/opentracing-go/log"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/sema"
+
+	"github.com/onflow/flow-go/module/trace"
+)
+
+const contractFunctionInvocationLocation = common.StringLocation("ContractFunctionInvocation")
+
+type TransactionContractFunctionInvocator struct {
+	contractLocation common.AddressLocation
+	functionName     string
+	arguments        []interpreter.Value
+	argumentTypes    []sema.Type
+	logger           zerolog.Logger
+}
+
+func NewTransactionContractFunctionInvocator(
+	contractLocation common.AddressLocation,
+	functionName string,
+	arguments []interpreter.Value,
+	argumentTypes []sema.Type,
+	logger zerolog.Logger) *TransactionContractFunctionInvocator {
+	return &TransactionContractFunctionInvocator{
+		contractLocation: contractLocation,
+		functionName:     functionName,
+		arguments:        arguments,
+		argumentTypes:    argumentTypes,
+		logger:           logger,
+	}
+}
+
+func (i *TransactionContractFunctionInvocator) Invoke(env *hostEnv, proc *TransactionProcedure) (cadence.Value, error) {
+	var span opentracing.Span
+	if env.ctx.Tracer != nil && proc.TraceSpan != nil {
+		span = env.ctx.Tracer.StartSpanFromParent(proc.TraceSpan, trace.FVMInvokeContractFunction)
+		span.LogFields(
+			traceLog.String("transaction.ContractFunctionCall", fmt.Sprintf("%s.%s", i.contractLocation.String(), i.functionName)),
```

Comment:
Minor: Could we prepare this once so we don't have to prepare it on each trace?

Reply:
I created a field on the TransactionContractFunctionInvocator to hold the LogFields